solr/core/src/java/org/apache/solr/core/NodeConfig.java

```diff
@@ -223,17 +221,23 @@ private NodeConfig(
 
     if (this.clusterSingletonPlugins != null
         && this.clusterSingletonPlugins.length > 0
```

Comment:
Do we need fields for clusterSingletonPlugins specifically?  Remember we're soon expanding this to other types like replica placement and hopefully more.  Could there be a Map of node/cluster/container plugins instead, in other words?  This way you're not singling out one specific type here in this check.